src/d/actor/d_a_btd.cpp

```diff
@@ -211,18 +2553,193 @@ static BOOL daBtd_IsDelete(btd_class*) {
 }
 
 /* 000081C0-00008304       .text daBtd_Delete__FP9btd_class */
-static BOOL daBtd_Delete(btd_class*) {
-    /* Nonmatching */
+static BOOL daBtd_Delete(btd_class* i_this) {
+    dComIfG_resDelete(&i_this->m02AC, "Btd");
+    dComIfG_resDelete(&i_this->m02B4, "Magma");
+    mDoHIO_deleteChild(l_HIO.mNo);
+    // !@bug: this for loop only iterates through the first 19 elements of the array.
+    for (int i = 0; i < 19; i++) {
+        mDoAud_seDeleteObject(&i_this->m5E98[i]);
+    }
+    for (s32 i = 0; i < (s32)ARRAY_SIZE(i_this->mBtdSibukiS); i++) {
+        mDoAud_seDeleteObject(&i_this->mBtdSibukiS[i].m04);
+    }
+    for (s32 i = 0; i < (s32)ARRAY_SIZE(i_this->m02FC); i++) {
+        mDoAud_seDeleteObject(&i_this->m02FC[i]);
+    }
+    if (dr2 != NULL) {
+        mDoAud_seDeleteObject(&dr2->unk_4A8);
+    }
+    mDoAud_seDeleteObject(&static_center_pos);
+    dKy_efplight_cut(&i_this->m60B0);
+    return TRUE;
 }
 
 /* 00008304-00008D30       .text useHeapInit__FP10fopAc_ac_c */
-static BOOL useHeapInit(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL useHeapInit(fopAc_ac_c* a_this) {
+    J3DModelData* modelDataA;
+    u32 uVar7;
+    J3DAnmTextureSRTKey* pJVar10;
+    int iVar11;
+    J3DAnmTevRegKey* pJVar13;
+    J3DModelData* modelDataB;
+    J3DModel* pJVar15;
+    f32 dVar18;
+
+    btd_class* i_this = (btd_class*)a_this;
+    i_this->m02BC = new mDoExt_McaMorf(
+        (J3DModelData*)dComIfG_getObjectRes("Btd", BTD_BMD_BTD),
+        NULL,
+        NULL,
+        (J3DAnmTransformKey*)dComIfG_getObjectRes("Btd", BTD_BCK_WAIT1),
+        J3DFrameCtrl::EMode_LOOP,
+        1.0f,
+        0,
+        -1,
+        0,
+        NULL,
+        0,
+        0x11020203
+    );
+    if ((i_this->m02BC == NULL) || (i_this->m02BC->getModel() == NULL)) {
+        uVar7 = 0;
+    } else {
+        for (u16 uVar16 = 0; uVar16 < i_this->m02BC->getModel()->getModelData()->getJointNum(); uVar16++) {
```

Comment:
The nesting is way too deep in this function, I'm pretty sure all these `uVar7 = } else {` parts should be early returns.
```suggestion
    if ((i_this->m02BC == NULL) || (i_this->m02BC->getModel() == NULL)) {
        return 0;
    }
    for (u16 uVar16 = 0; uVar16 < i_this->m02BC->getModel()->getModelData()->getJointNum(); uVar16++) {
```